http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -848,6 +871,103 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
+    implicit ev: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Installing helm chart ${config.customAppConfig.chart} for app ${appName.value} in cluster ${dbCluster.getGkeClusterId.toString}"
+      )
+
+      desc <- F.fromOption(descriptorOpt, AppRequiresDescriptorException(appId))
+
+      _ <- logger.info(ctx.loggingCtx)(
+        s"about to process descriptor for app ${appName.value} in cluster ${dbCluster.getGkeClusterId.toString} | trace id: ${ctx.traceId}"
+      )
+
+      descriptor <- appDescriptorDAO.getDescriptor(desc).adaptError {
+        case e =>
+          AppCreationException(
+            s"Failed to process descriptor: $desc. Please ensure it is a valid descriptor, and that the remote file is valid yaml following the schema detailed here: https://github.com/DataBiosphere/terra-app#app-schema. \n\tOriginal message: ${e.getMessage}"
+          )
+      }
+
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Finished processing descriptor for app ${appName.value} in cluster ${dbCluster.getGkeClusterId.toString} | trace id: ${ctx.traceId}"
+      )
+
+      // TODO we're only handling 1 service for now
+      (serviceName, serviceConfig) = descriptor.services.head
+
+      // Save the service in the DB
+      _ <- serviceQuery
+        .saveForApp(
+          appId,
+          KubernetesService(
+            ServiceId(-1),
+            ServiceConfig(ServiceName(serviceName),
+                          org.broadinstitute.dsde.workbench.leonardo.KubernetesServiceKindName("ClusterIP"))
+          )
+        )
+        .transaction
+
+      helmAuthContext <- getHelmAuthContext(googleCluster, dbCluster, namespaceName)
+
+      chartValues = buildCustomChartOverrideValuesString(appName,
+                                                         release,
+                                                         nodepoolName,
+                                                         serviceName,
+                                                         dbCluster,
+                                                         namespaceName,
+                                                         serviceConfig,
+                                                         extraArgs,
+                                                         disk,
+                                                         serviceConfig.environment ++ customEnvironmentVariables)
+
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Chart override values are: ${chartValues} | trace id: ${ctx.traceId}"
+      )
+
+      // Invoke helm
+      _ <- helmClient
+        .installChart(
+          release,
+          config.customAppConfig.chartName,
+          config.customAppConfig.chartVersion,
+          org.broadinstitute.dsp.Values(chartValues)
+        )
+        .run(helmAuthContext)
+
+      // Poll app until it starts up
+      last <- streamFUntilDone(
```

Comment:
can probably just use `streamUntilDoneOrTimeout`

Reply:
I kinda like `streamFUntilDone` so we can check the `Option` below and throw `AppCreationException`